hudi-client/src/main/java/org/apache/hudi/func/bulkinsert/BulkInsertInternalPartitioner.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.func.bulkinsert;
+
+import org.apache.hudi.common.model.HoodieRecordPayload;
+import org.apache.hudi.table.UserDefinedBulkInsertPartitioner;
+
+public abstract class BulkInsertInternalPartitioner<T extends HoodieRecordPayload> implements
+    UserDefinedBulkInsertPartitioner<T> {
```

Comment:
We can rename UserDefinedBulkInsertPartitioner to just `BulkInsertPartitioner` and reuse it I think?

Reply:
RDD API clients i.e Uber/marmaray  may need to change the name of the interface implemented. It should be fine IMO. cc @bvaradar @n3nash

Reply:
Yeah.  I was also wondering about the custom implementation of `UserDefinedBulkInsertPartitioner` in my previous comment..

Reply:
Can we just keep the UserDefinedBulkInsertPartitioner as an interface and then provide an abstract class here ? Might be fine to change at Uber, but it might be best to avoid a breaking change since it's simple in this case ?

Reply:
@vinothchandar @n3nash @bvaradar Let's have a consensus on whether to change the `UserDefinedBulkInsertPartitioner` interface.  I'm fine with either way.  Right now I've also changed the interface so it will be a breaking change.

Reply:
I was trying to generalize the existing interface, since what we are doing here is not userdefined anyway.. 

>but it might be best to avoid a breaking change since it's simple in this case ?

this is also fair.. so we can leave the current interface as is and you can just extend/implement it.